collectors/python.d.plugin/samba/samba.chart.py

```diff
@@ -107,23 +108,25 @@ def __init__(self, configuration=None, name=None):
         self.rgx_smb2 = re.compile(r'(smb2_[^:]+|syscall_.*file_bytes):\s+(\d+)')
 
     def check(self):
-        sudo_binary = find_binary(SUDO)
-        if not sudo_binary:
-            self.error("can't locate '{0}' binary".format(SUDO))
-            return False
-
         smbstatus_binary = find_binary(SMBSTATUS)
         if not smbstatus_binary:
             self.error("can't locate '{0}' binary".format(SMBSTATUS))
             return False
 
+        if os.getuid() == 0:
```

Comment:
this should be in the beginning, there is no need to check for `smbstatus_binary` if uid == 0

Reply:
🙈